include/VecCore/Backend/Implementation.h

```diff
@@ -284,6 +285,21 @@ VECCORE_FORCE_INLINE VECCORE_ATT_HOST_DEVICE constexpr Bool_s EarlyReturnAllowed
 #endif
 }
 
+// An extension of EarlyReturnAllowed(), and new name is a reminder that it can be used for other purposes as well
+template <typename T, size_t simdSize = 1>
+VECCORE_FORCE_INLINE
+VECCORE_ATT_HOST_DEVICE
+constexpr Bool_s SIMDsizeUpTo()
```

Comment:
> Sure, but your logic is a bit backwards. SIMDSizeAtLeast<T, N>() 

How so?  Using SIMDSizeAtLeast would require all the usage to have a not (!) in front.  I.e. the usage is if  SIMDSizeAtMost<n> && EarlyReturnAllowed() then do early return.

> and there is no performance penalty, because the first if is evaluated at compile time.

There is no penalty in either case since the function is a constexpr.

I just remember another reason why @mrguilima prefered to have a function.  The reason is that one would want to have an **easy** to disable this **kind** of early return and having a single function that control it helps.